### DISCUSSION TITLE
Custom api/extension for Clusters

### DISCUSSION THREAD

**Discussion by kbocock-krg** [Dynatrace/dynatrace-configuration-as-code#792] [Ideas]
Currently we work within a kube cluster. After reading the api docs and how to add custom apis I believe there are some more features that would benefit users. 

1. I would provide a mechanism to query a cluster---> Cloud app from the cluster name. Ex. Prometheus Stack is running on a cluster in n environments. I want the same alert setup for each. 
2. Allow the user to extend custom apis at runtime. Check for a folder with an expected name and allow the user to add in a single json file the path etc. necessary to leverage another api like the entity filter.

**Comment by UnseenWizzard** [comment]
1) As mentioned on your other question we will offer a way to reference existing entities in a similar format to the entity selector API (https://www.dynatrace.com/support/help/dynatrace-api/environment-api/entity-v2/entity-selector) 

However for alerting we have internally and with customers found it to be best practice to use auto-tagging rules to ensure all the applications, services, etc you wish to be covered receive the same tag, and the limit alerting profiles to the tag rather than entities. 

I think this practice should already work for your usecase.

2) While that feature did exist in the past the number of "easy to add" APIs that could be covered by simple configuration is currently (close to) zero, with every not yet supported API needing at least some code-changes to support. 
That being said, we are always happy about contributions so please do not hesitate to open a Pull Request. 

For entity selectors however we do have a plan on how we will support this, so possibly just wait for a future release
